### PR TITLE
Feature/dei 229 depth averaged rules sigma models

### PR DESCRIPTION
**Pull request by mKlapwijk**

### CODE REVIEW

File: decoimpact/business/entities/rule_based_model.py [hunk 1]
```diff
@@ -207,16 +209,20 @@ def _validate_mappings(self, mappings: dict[str, str], logger: ILogger) -> bool:
...
 
+        Extend variable names specific for Delft3D based on dummy variable name.
+
         Returns:
             Dict[str, List[str]]
         """
         rule_input_vars = [rule.input_variable_names for rule in self._rules]
         rule_output_vars = [rule.output_variable_name for rule in self._rules]
 
+        rule_input_vars = [list(_du.extend_to_full_name(rule_input_vars[0], dummy_variable))]
```
Comment (CindyvdVries) [inline decoimpact/business/entities/rule_based_model.py:224]: is already a list? do you need to cast it here? why rule_input_vars**[0]**?

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 2]
```diff
@@ -46,10 +47,14 @@ def execute(
         # just used the first value.
         variables = next(iter(value_arrays.values()))
 
+        #dummy_var = get_dummy_variable_in_ugrid(self.dataset)
+        dummy_var = "mesh2d"
+        print("Q",dummy_var)
```
[the remark this answers is not in the document]
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py:52]: Yes I know, this part still needs to be done

File: decoimpact/business/entities/rules/rule_base.py [hunk 3]
```diff
@@ -53,6 +53,11 @@ def input_variable_names(self) -> List[str]:
         """Name of the input variable"""
         return self._input_variable_names
 
+    @input_variable_names.setter
+    def input_variable_names(self, input_variable_names: str) -> List[str]:
```
Comment (CindyvdVries) [inline decoimpact/business/entities/rules/rule_base.py:57]:
```suggestion
    def input_variable_names(self, input_variable_names: List[str]) -> List[str]:
```

File: decoimpact/data/parsers/parser_depth_average_rule.py [hunk 4]
```diff
@@ -12,16 +12,16 @@
 """
 from typing import Any, Dict, List
 
+from decoimpact.crosscutting.delft3d_specific_data import (
+    BED_LEVEL_SUFFIX,
```
Comment (CindyvdVries) [inline decoimpact/data/parsers/parser_depth_average_rule.py:16]: everywhere the order of these is different :'(

File: tests/business/utils/test_dataset_utils.py [hunk 5]
```diff
@@ -328,20 +328,21 @@ def test_get_dummy_and_dependent_var_list(self):
         """Test if you receive the name of the dummy and dependent variables variable
         in a ugrid dataset"""
         # Arrange
-        var_list = ["var1", "var2", "var3", "var4", "var5"]
+        var_list = ["mesh2d", "var2", "var3", "var4", "var5"]
```
Comment (CindyvdVries) [inline tests/business/utils/test_dataset_utils.py:331]: it should not matter what the name of the dummy var is right?

**Review by CindyvdVries — CHANGES_REQUESTED** [review]
some minor things

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 6]
```diff
@@ -35,21 +35,23 @@ def execute(
...
+            value_array (DataArray): Values to average over the depth
 
         Returns:
-            DataArray: Averaged values
+            DataArray: Depth-averaged values
         """
 
         # The first DataArray in our value_arrays contains the values to be averaged
         # but the name of the key is given by the user, and is unknown here, so
         # just used the first value.
```
[the remark this answers is not in the document]
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py:46]: done

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 7]
```diff
@@ -106,3 +108,21 @@ def execute(
         return relative_values.sum(dim=dim_layer_name) / layer_heights.sum(
             dim=dim_layer_name
         )
+
+    def _extract_variable_based_on_suffix(
```
[the remark this answers is not in the document]
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py:112]: done

File: decoimpact/data/parsers/parser_depth_average_rule.py (hunk 4 above)
Comment (mKlapwijk) [reply decoimpact/data/parsers/parser_depth_average_rule.py:16]: not anymore!

File: tests/business/utils/test_dataset_utils.py (hunk 5 above)
Comment (mKlapwijk) [reply tests/business/utils/test_dataset_utils.py:331]: no it does'nt, but this made it easier to check if it was correct

File: decoimpact/business/utils/dataset_utils.py [hunk 8]
```diff
@@ -374,6 +372,7 @@ def extend_to_full_name(
     Returns:
         list[str]: list of the extended variable names
     """
+    dummy_variable = dummy_variable[0]
```
Comment (CindyvdVries) [inline decoimpact/business/utils/dataset_utils.py:375]: if dummy_variable comes from list ([0]) is the type str (line 363 correct?)

File: decoimpact/business/entities/rules/depth_average_rule.py [hunk 9]
```diff
@@ -113,7 +113,7 @@ def _extract_variable_based_on_suffix(
             self,
             value_arrays: Dict[str, _xr.DataArray],
             suffix: str
-            ):
+            ) -> list:
```
Comment (CindyvdVries) [inline decoimpact/business/entities/rules/depth_average_rule.py:116]: List (import from Typing)

File: decoimpact/business/utils/dataset_utils.py [hunk 10]
```diff
@@ -222,26 +222,24 @@ def get_dummy_variable_in_ugrid(dataset: _xr.Dataset) -> list:
...
-    in one list.
+    The variables are
+    recursively looked up based on the dummy variable.
     This is done to support XUgrid and to prevent invalid topologies.
     This also allows QuickPlot to visualize the results.
 
     Args:
         dataset (_xr.Dataset): Dataset to search for dummy variable
-
+        dummy_vars (list[str]): dummy variables
```
Comment (CindyvdVries) [inline decoimpact/business/utils/dataset_utils.py:234]: List (also line 237)

**Review by CindyvdVries — APPROVED** [review]
Small last things, but otherwise Good to Merge! :)

File: decoimpact/business/utils/dataset_utils.py (hunk 10 above)
Comment (mKlapwijk) [reply decoimpact/business/utils/dataset_utils.py:234]: done

File: decoimpact/business/entities/rules/depth_average_rule.py (hunk 9 above)
Comment (mKlapwijk) [reply decoimpact/business/entities/rules/depth_average_rule.py:116]: done

File: decoimpact/business/utils/dataset_utils.py (hunk 8 above)
Comment (mKlapwijk) [reply decoimpact/business/utils/dataset_utils.py:375]: no it was not, should be a List[str]. Changed

**Pull request merged**